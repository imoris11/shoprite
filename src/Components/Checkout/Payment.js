import React, {Component} from 'react';
import StripeCheckout from './Form';
import { Row, Col } from 'react-bootstrap';
import '../../Resources/css/payment.css';
import { getTotalCostApi } from './CheckoutApi';
import LinearIndeterminate from '../Extra/Loading';
export default class Payment extends Component {
  state = {
    payment:'card',
    total:0,
  }
  componentDidMount () {
    this.getTotalCost();
  }
  async getTotalCost () {
    let { total } = this.state;
    //Get total cost from online cart
    let cart_id = localStorage.getItem('cart_id');
    let url = 'https://backendapi.turing.com/shoppingcart/totalAmount/'+cart_id;
    let res = await getTotalCostApi(url);
    total = Number(res.total_amount);

    this.setState({total});
  }
  onNext = () => this.props.onNext();
  setIsLoading = () => this.setState({loading:true})
  render () {
    let {payment} = this.state;
    return (
      <Row>
        <Col md={{span:8, offset:2}}>
          <Row>
            <Col onClick={()=>this.setState({payment:'card'})} className='text-center payment-method' style={{backgroundColor: payment === 'card' && 'lightblue'}} >
              <img src={require('../../Resources/images/logos-visa.png')} className='payment-image' alt="Visa" />
              <img src={require('../../Resources/images/logos-mastercard.png')} className='payment-image' alt="Mastercard" />
            </Col>
            <Col onClick={()=>this.setState({payment:'paypal'})} className='text-center payment-method' style={{backgroundColor: payment === 'paypal' && 'lightblue'}}>
              <img src={require('../../Resources/images/logos-PayPal.png')} className='payment-image' alt="Paypal" />
            </Col>
          </Row>
          {!this.state.loading && <Col>
            <h6 >Total: ${this.state.total.toFixed(2)}</h6>
            {payment === 'card' &&
            <StripeCheckout isLoading={this.setIsLoading} onNext={this.onNext} />}
          </Col>}
          {this.state.loading && <LinearIndeterminate />}
        </Col>
      </Row>
    )
  }
}
