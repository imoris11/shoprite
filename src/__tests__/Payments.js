import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import Payment from '../Components/Checkout/Payment';

describe('Test Payment Component', () => {

   it('renders without crashing', () => {
      shallow(<Payment />);
    });

    it('renders correctly', ()=> {
      const component = shallow(<Payment />);
      expect(component).toMatchSnapshot();
    });

});
