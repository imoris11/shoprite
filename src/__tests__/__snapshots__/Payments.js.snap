// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Payment Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Payment />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        as="div"
        md={
          Object {
            "offset": 2,
            "span": 8,
          }
        }
      >
        <ForwardRef(render)>
          <ForwardRef
            as="div"
            className="text-center payment-method"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          >
            <img
              alt="Visa"
              className="payment-image"
              src="logos-visa.png"
            />
            <img
              alt="Mastercard"
              className="payment-image"
              src="logos-mastercard.png"
            />
          </ForwardRef>
          <ForwardRef
            as="div"
            className="text-center payment-method"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": false,
              }
            }
          >
            <img
              alt="Paypal"
              className="payment-image"
              src="logos-PayPal.png"
            />
          </ForwardRef>
        </ForwardRef(render)>
        <ForwardRef
          as="div"
        >
          <h6>
            SubTotal: $
            0.00
          </h6>
          <h6>
            Tax: $
            0.00
          </h6>
          <h6>
            Shipping: $
            0.00
          </h6>
          <h5>
            Total: $
            0.00
          </h5>
          <Form
            isLoading={[Function]}
            onNext={[Function]}
          />
        </ForwardRef>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": Array [
          <ForwardRef(render)>
            <ForwardRef
              as="div"
              className="text-center payment-method"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            >
              <img
                alt="Visa"
                className="payment-image"
                src="logos-visa.png"
              />
              <img
                alt="Mastercard"
                className="payment-image"
                src="logos-mastercard.png"
              />
            </ForwardRef>
            <ForwardRef
              as="div"
              className="text-center payment-method"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": false,
                }
              }
            >
              <img
                alt="Paypal"
                className="payment-image"
                src="logos-PayPal.png"
              />
            </ForwardRef>
          </ForwardRef(render)>,
          <ForwardRef
            as="div"
          >
            <h6>
              SubTotal: $
              0.00
            </h6>
            <h6>
              Tax: $
              0.00
            </h6>
            <h6>
              Shipping: $
              0.00
            </h6>
            <h5>
              Total: $
              0.00
            </h5>
            <Form
              isLoading={[Function]}
              onNext={[Function]}
            />
          </ForwardRef>,
          undefined,
        ],
        "md": Object {
          "offset": 2,
          "span": 8,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                as="div"
                className="text-center payment-method"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "lightblue",
                  }
                }
              >
                <img
                  alt="Visa"
                  className="payment-image"
                  src="logos-visa.png"
                />
                <img
                  alt="Mastercard"
                  className="payment-image"
                  src="logos-mastercard.png"
                />
              </ForwardRef>,
              <ForwardRef
                as="div"
                className="text-center payment-method"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": false,
                  }
                }
              >
                <img
                  alt="Paypal"
                  className="payment-image"
                  src="logos-PayPal.png"
                />
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": Array [
                  <img
                    alt="Visa"
                    className="payment-image"
                    src="logos-visa.png"
                  />,
                  <img
                    alt="Mastercard"
                    className="payment-image"
                    src="logos-mastercard.png"
                  />,
                ],
                "className": "text-center payment-method",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "lightblue",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Visa",
                    "className": "payment-image",
                    "src": "logos-visa.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Mastercard",
                    "className": "payment-image",
                    "src": "logos-mastercard.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": <img
                  alt="Paypal"
                  className="payment-image"
                  src="logos-PayPal.png"
                />,
                "className": "text-center payment-method",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": false,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Paypal",
                  "className": "payment-image",
                  "src": "logos-PayPal.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <h6>
                SubTotal: $
                0.00
              </h6>,
              <h6>
                Tax: $
                0.00
              </h6>,
              <h6>
                Shipping: $
                0.00
              </h6>,
              <h5>
                Total: $
                0.00
              </h5>,
              <Form
                isLoading={[Function]}
                onNext={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "SubTotal: $",
                  "0.00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "SubTotal: $",
                "0.00",
              ],
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Tax: $",
                  "0.00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Tax: $",
                "0.00",
              ],
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Shipping: $",
                  "0.00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Shipping: $",
                "0.00",
              ],
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Total: $",
                  "0.00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Total: $",
                "0.00",
              ],
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isLoading": [Function],
                "onNext": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
        },
        "displayName": "Col",
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": undefined,
      "displayName": "Bootstrap(Row)",
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          as="div"
          md={
            Object {
              "offset": 2,
              "span": 8,
            }
          }
        >
          <ForwardRef(render)>
            <ForwardRef
              as="div"
              className="text-center payment-method"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            >
              <img
                alt="Visa"
                className="payment-image"
                src="logos-visa.png"
              />
              <img
                alt="Mastercard"
                className="payment-image"
                src="logos-mastercard.png"
              />
            </ForwardRef>
            <ForwardRef
              as="div"
              className="text-center payment-method"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": false,
                }
              }
            >
              <img
                alt="Paypal"
                className="payment-image"
                src="logos-PayPal.png"
              />
            </ForwardRef>
          </ForwardRef(render)>
          <ForwardRef
            as="div"
          >
            <h6>
              SubTotal: $
              0.00
            </h6>
            <h6>
              Tax: $
              0.00
            </h6>
            <h6>
              Shipping: $
              0.00
            </h6>
            <h5>
              Total: $
              0.00
            </h5>
            <Form
              isLoading={[Function]}
              onNext={[Function]}
            />
          </ForwardRef>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": Array [
            <ForwardRef(render)>
              <ForwardRef
                as="div"
                className="text-center payment-method"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "lightblue",
                  }
                }
              >
                <img
                  alt="Visa"
                  className="payment-image"
                  src="logos-visa.png"
                />
                <img
                  alt="Mastercard"
                  className="payment-image"
                  src="logos-mastercard.png"
                />
              </ForwardRef>
              <ForwardRef
                as="div"
                className="text-center payment-method"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": false,
                  }
                }
              >
                <img
                  alt="Paypal"
                  className="payment-image"
                  src="logos-PayPal.png"
                />
              </ForwardRef>
            </ForwardRef(render)>,
            <ForwardRef
              as="div"
            >
              <h6>
                SubTotal: $
                0.00
              </h6>
              <h6>
                Tax: $
                0.00
              </h6>
              <h6>
                Shipping: $
                0.00
              </h6>
              <h5>
                Total: $
                0.00
              </h5>
              <Form
                isLoading={[Function]}
                onNext={[Function]}
              />
            </ForwardRef>,
            undefined,
          ],
          "md": Object {
            "offset": 2,
            "span": 8,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  as="div"
                  className="text-center payment-method"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "lightblue",
                    }
                  }
                >
                  <img
                    alt="Visa"
                    className="payment-image"
                    src="logos-visa.png"
                  />
                  <img
                    alt="Mastercard"
                    className="payment-image"
                    src="logos-mastercard.png"
                  />
                </ForwardRef>,
                <ForwardRef
                  as="div"
                  className="text-center payment-method"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": false,
                    }
                  }
                >
                  <img
                    alt="Paypal"
                    className="payment-image"
                    src="logos-PayPal.png"
                  />
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "div",
                  "children": Array [
                    <img
                      alt="Visa"
                      className="payment-image"
                      src="logos-visa.png"
                    />,
                    <img
                      alt="Mastercard"
                      className="payment-image"
                      src="logos-mastercard.png"
                    />,
                  ],
                  "className": "text-center payment-method",
                  "onClick": [Function],
                  "style": Object {
                    "backgroundColor": "lightblue",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Visa",
                      "className": "payment-image",
                      "src": "logos-visa.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Mastercard",
                      "className": "payment-image",
                      "src": "logos-mastercard.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                  },
                  "displayName": "Col",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "div",
                  "children": <img
                    alt="Paypal"
                    className="payment-image"
                    src="logos-PayPal.png"
                  />,
                  "className": "text-center payment-method",
                  "onClick": [Function],
                  "style": Object {
                    "backgroundColor": false,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Paypal",
                    "className": "payment-image",
                    "src": "logos-PayPal.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                  },
                  "displayName": "Col",
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "Bootstrap(Row)",
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": Array [
                <h6>
                  SubTotal: $
                  0.00
                </h6>,
                <h6>
                  Tax: $
                  0.00
                </h6>,
                <h6>
                  Shipping: $
                  0.00
                </h6>,
                <h5>
                  Total: $
                  0.00
                </h5>,
                <Form
                  isLoading={[Function]}
                  onNext={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "SubTotal: $",
                    "0.00",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "SubTotal: $",
                  "0.00",
                ],
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Tax: $",
                    "0.00",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Tax: $",
                  "0.00",
                ],
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Shipping: $",
                    "0.00",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Shipping: $",
                  "0.00",
                ],
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Total: $",
                    "0.00",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Total: $",
                  "0.00",
                ],
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "isLoading": [Function],
                  "onNext": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          undefined,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "Bootstrap(Row)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
